tensorflow_addons/metrics/harmonic_mean.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Implements HarmonicMean."""
+
+import tensorflow as tf
+
+from typeguard import typechecked
+from tensorflow_addons.utils.types import AcceptableDTypes
+from tensorflow_addons.metrics.utils import sample_weight_shape_match
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class HarmonicMean(tf.keras.metrics.Mean):
+    """Compute Harmonic Mean
+    The harmonic mean is a kind of mean. It can be expressed as the reciprocal of
+    the arithmetic mean of the reciprocals of the given set of numbers.
+    Note: `tfa.metrics.HarmonicMean` can be used the same as `tf.keras.metrics.Mean`.
+    Args:
+        name: (Optional) String name of the metric instance.
+        dtype: (Optional) Data type of the metric result.
+    Usage:
+    >>> metric = tfa.metrics.HarmonicMean()
+    >>> metric.update_state([1, 4, 4])
+    >>> metric.result().numpy()
+    2.0
+    """
+
+    @typechecked
+    def __init__(
+        self, name: str = "harmonic_mean", dtype: AcceptableDTypes = None, **kwargs
+    ):
+        super().__init__(name=name, dtype=dtype, **kwargs)
+
+    def update_state(self, values, sample_weight=None) -> None:
+        values = tf.cast(values, dtype=self.dtype)
+        sample_weight = sample_weight_shape_match(values, sample_weight)
+        sample_weight = tf.cast(sample_weight, dtype=self.dtype)
+        super().update_state(tf.math.reciprocal(values), sample_weight)
```

Comment:
Do we need to filter with `reciprocal_nan` here or mask `tf.boolean_mask(x, tf.math.is_finite(values))`?

Reply:
Do you mean `reciprocal_no_nan`?

Assuming that, yeah I think that would make more sense.

Reply:
Yes and masking for infinite

Reply:
Looking at the `scipy.stats.hmean's` behavior, infinite values should be allowed IMO. And about using `tf.reciprocal_no_nan`, I realized that doing that would turn into unexpected results when there are 0's in the input data.

Reply:
Pushed it without `reciprocal_no_nan` and `masking`. Let me know if I need to change something.